src/opendr/perception/pose_estimation/hr_pose_estimation/high_resolution_learner.py

```diff
@@ -0,0 +1,610 @@
+            avoid cropping images with abnormal ratios e.g (30, 800)
+        :type max_width: int
+        :param upsample_ratio: Defines the amount of upsampling to be performed on the heatmaps and PAFs when resizing,
+            defaults to 4
+        :type upsample_ratio: int, optional
+
+         :returns: the heatmap of human figures, the part affinity filed (pafs), the scale of the resized image compred
+            to the initial and the pad arround the image
+         :rtype: heatmap, pafs -> numpy.ndarray
+                 scale -> float
+                 pad = -> list
+        """
+
+        height, width, _ = img.shape
+        scale = net_input_height_size / height
+        img_ratio = width / height
+        if img_ratio > 6:
+            scale = max_width / width
+
+        scaled_img = cv2.resize(img, (0, 0), fx=scale, fy=scale, interpolation=cv2.INTER_LINEAR)
+        scaled_img = normalize(scaled_img, img_mean, img_scale)
+        min_dims = [net_input_height_size, max(scaled_img.shape[1], net_input_height_size)]
+        padded_img, pad = pad_width(scaled_img, stride, pad_value, min_dims)
+
+        if 'cuda' in self.device:
+            tensor_img = torch.from_numpy(padded_img).permute(2, 0, 1).unsqueeze(0).float().cuda()
+            tensor_img = tensor_img.cuda()
+            if self.half:
+                tensor_img = tensor_img.half()
+        else:
+            tensor_img = torch.from_numpy(padded_img).permute(2, 0, 1).unsqueeze(0).float().cpu()
+
+        stages_output = net(tensor_img)
+
+        stage2_heatmaps = stages_output[-2]
+        heatmaps = np.transpose(stage2_heatmaps.squeeze().cpu().data.numpy(), (1, 2, 0))
+        heatmaps = heatmaps.astype(np.float32)
+        heatmaps = cv2.resize(heatmaps, (0, 0), fx=upsample_ratio, fy=upsample_ratio, interpolation=cv2.INTER_CUBIC)
+
+        stage2_pafs = stages_output[-1]
+        pafs = np.transpose(stage2_pafs.squeeze().cpu().data.numpy(), (1, 2, 0))
+        pafs = pafs.astype(np.float32)
+        pafs = cv2.resize(pafs, (0, 0), fx=upsample_ratio, fy=upsample_ratio, interpolation=cv2.INTER_CUBIC)
+
+        return heatmaps, pafs, scale, pad
+
+    def __pooling(self, img, kernel):  # Pooling on input image for dimension reduction
+        """This method applies a pooling filter on an input image in order to resize it in a fixed shape
+
+         :param img: input image for resizing
+         :rtype img: engine.data.Image class object
+         :param kernel: the kernel size of the pooling filter
+         :type kernel: int
+         """
+        pool_img = torchvision.transforms.ToTensor()(img)
+        pool_img = pool_img.unsqueeze(0)
+        pool_img = torch.nn.functional.avg_pool2d(pool_img, kernel)
+        pool_img = pool_img.squeeze(0).permute(1, 2, 0).cpu().float().numpy()
+        return pool_img
+
+    def fit(self, dataset, val_dataset=None, logging_path='', silent=True, verbose=True):
```

Comment:
Since the parent class changed, the overridden method  needs to match the parent's signature. This follows for the other methods too.
```suggestion
    def fit(self, dataset, val_dataset=None, logging_path='', logging_flush_secs=30,
            silent=False, verbose=True, epochs=None, use_val_subset=True, val_subset_size=250,
            images_folder_name="train2017", annotations_filename="person_keypoints_train2017.json",
            val_images_folder_name="val2017", val_annotations_filename="person_keypoints_val2017.json"):
```